conans/client/tools/win.py

```diff
@@ -437,42 +450,10 @@ def vcvars_command(settings, arch=None, compiler_version=None, force=False, vcva
 
 def vcvars_dict(settings, arch=None, compiler_version=None, force=False, filter_known_paths=False,
                 vcvars_ver=None, winsdk_version=None, only_diff=True, output=None):
-    known_path_lists = ("include", "lib", "libpath", "path")
     cmd = vcvars_command(settings, arch=arch,
                          compiler_version=compiler_version, force=force,
                          vcvars_ver=vcvars_ver, winsdk_version=winsdk_version, output=output)
-    cmd += " && set"
-    ret = check_output_runner(cmd)
-    new_env = {}
-    for line in ret.splitlines():
-        line = line.strip()
-
-        if line == "\n" or not line:
-            continue
-        try:
-            name_var, value = line.split("=", 1)
-            new_value = value.split(os.pathsep) if name_var.lower() in known_path_lists else value
-            # Return only new vars & changed ones, but only with the changed elements if the var is
-            # a list
-            if only_diff:
-                old_value = os.environ.get(name_var)
-                if name_var.lower() == "path":
-                    old_values_lower = [v.lower() for v in old_value.split(os.pathsep)]
-                    # Clean all repeated entries, not append if the element was already there
-                    new_env[name_var] = [v for v in new_value if v.lower() not in old_values_lower]
-                elif old_value and value.endswith(os.pathsep + old_value):
-                    # The new value ends with separator and the old value, is a list,
-                    # get only the new elements
-                    new_env[name_var] = value[:-(len(old_value) + 1)].split(os.pathsep)
-                elif value != old_value:
-                    # Only if the vcvars changed something, we return the variable,
-                    # otherwise is not vcvars related
-                    new_env[name_var] = new_value
-            else:
-                new_env[name_var] = new_value
-
-        except ValueError:
-            pass
+    new_env = env_diff(cmd, only_diff)
```

Comment:
although this might imply duplicating the code, maybe we have to keep this unchanged to avoid any kind on breaking behavior cc/ @memsharded

Reply:
is there a benefit? as we have tests for vcvars, as long as tests pass, we should be absolutely confident to change the implementation, right?

Reply:
We are changing the variables that this `env_diff()` is gathering. Sure the tests are a good proof, but they may not cover all the cases. We have to be extra careful with this as the functionality of vcvars is widely used

Reply:
okay, I'll keep it as is then